foolbox/attacks/fast_gradient_method.py

```diff
@@ -28,6 +46,20 @@ def __init__(self, *, random_start: bool = False):
             rel_stepsize=1.0, steps=1, random_start=random_start,
         )
 
+    def run(  # type: ignore
```

Comment:
This was necessary as otherwise, the type checks would have failed because the type of the `criterion` argument is different in the superclass (`Union[Missclassification, TargetedMisclassification, T]` vs `Union[Missclassification, T]`); however, I think that's not best practice but all other solutions I could think of would make the code more complex to read.